aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -196,15 +206,28 @@ Tensor stft(const Tensor& self, const int64_t n_fft, const optional<int64_t> hop
     } else { \
       SS << "None"; \
     } \
-    SS << ", normalized=" << normalized << ", onesided=" << onesided << ")"
+    SS << ", normalized=" << normalized << ", onesided="; \
+    write_opt(SS, onesidedOpt) << ", return_complex="; \
+    write_opt(SS, return_complexOpt) << ") "
 
   // default_init hop_length and win_length
   auto hop_length = hop_lengthOpt.value_or(n_fft >> 2);
   auto win_length = win_lengthOpt.value_or(n_fft);
+  const bool return_complex = return_complexOpt.value_or(
+      self.is_complex() || (window.defined() && window.is_complex()));
+  if (!return_complexOpt && !return_complex) {
+    TORCH_WARN("stft will return complex tensors by default in future, use"
```

Comment:
I appreciate you've been very thoughtful about how to deprecate the current functionality. To step back for a moment, however, what if we left the current functionality as is? That is, what if float -> float[2] was kept and only the complex functionality was added?

Reply:
Well `librosa` always returns a complex tensor, so this is the right direction for `librosa` compat. Existing users can always set `return_complex=False` to keep the old behaviour.